#40 testing

MMCQ33 opened the pull request:
Added More testing

MMCQ33 committed:
MindBuilderTest1.0 made

Added first couple of tests, made things accesible for testing.

MMCQ33 committed:
Only execute when agent is on navmesh

MMCQ33 committed:
Merge branch 'Develop' into #40-Testing

MMCQ33 committed:
made equals function

MMCQ33 committed:
Added basic testing for gathering

Added testing for gathering

MMCQ33 committed:
Finished Basic tests

Finished Tests for base building. Now tests the functionality of making and splitting ant rooms.

MMCQ33 committed:
small changes that did not get pushed before

MMCQ33 committed:
Updated Paused Setting

Updated paused so you dont have to press escape twice

MMCQ33 committed:
Made it  bit more pretty

Added more setters and getters to make it more closed

MMCQ33 committed:
removed test scenes

removed test scenes and refined the code

MMCQ33 committed:
updated Scouting test

Made scouting test wait for frames instead of a time.

MMCQ33 committed:
Ignore Test Scenes

ignore test scenes made by unity

MMCQ33 committed:
Made the tests more reliable

Changed some yields to make the tests more reliable. Also made them work without focus on the game window.

MMCQ33 committed:
Made it work with private resources

Made it work if known and unknown resources are private.

MMCQ33 committed:
Merge branch 'Develop' into #40-Testing

MMCQ33 committed:
fixed merge issues

fixed issues caused by merge. Baked navmesh again.

ErikLouwerse commented on Hivemind/Assets/BaseBuildingTestScene.unity:
```diff
@@ -0,0 +1,6122 @@
+%YAML 1.1
```
Please put this in the scenes folder

ErikLouwerse commented on Hivemind/Assets/Scenes/BaseBuildingTestScene 1.meta:
```diff
@@ -0,0 +1,8 @@
+fileFormatVersion: 2
```
What meta file is this for?

ErikLouwerse commented on Hivemind/Assets/Scripts/Ant.cs:
```diff
@@ -183,6 +198,11 @@ private void ChangeGroupID(object sender, GroupIdChangedEventArgs e)
         }
     }
 
+    public List<IMind> GetMinds()
```
It's already public.
Please remove or make public set function and private the variable

ErikLouwerse commented on Hivemind/Assets/Scripts/MindGroup.cs:
```diff
@@ -20,7 +20,7 @@ public MindGroup(GameObject UiObject)
     public int Count { get; private set; }
 
-    public List<IMind> Minds { get; }
+    public List<IMind> Minds { get; set; }
```
pls remove getset as it is useless

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/Gathering.cs:
```diff
@@ -238,7 +241,13 @@ private void TargetResource()
         if (target != null)
         {
-            if (state == State.Idle) ant.GetAgent().isStopped = false;
+            if (state == State.Idle)
```
just make double if

ErikLouwerse commented on Hivemind/Assets/Scripts/Minds/Gathering.cs:
```diff
@@ -290,9 +302,35 @@ private void carryResource(ResourceNode resource)
 
     private IEnumerator Scout()
+    {
+       
```
WHITESPACE

ErikLouwerse commented on Hivemind/Assets/Tests/PlayModeTests/BaseBuildingTest.cs:
```diff
@@ -0,0 +1,99 @@
+    {
+        private GameObject gameUI;
+        private UiController uiController;
+        private UnitController unitControl;
```
Useless

ErikLouwerse commented on Hivemind/Assets/Tests/PlayModeTests/BaseBuildingTest.cs:
```diff
@@ -0,0 +1,99 @@
+        [UnitySetUp]
+        public IEnumerator Init()
+        {
+            SceneManager.LoadScene("BaseBuildingTestScene");
+            yield return new WaitForSeconds(1f);
```
```suggestion
            AsyncOperation asyncLoadLevel = SceneManager.LoadSceneAsync("BaseBuildingTestScene", LoadSceneMode.Single);
            while (!asyncLoadLevel.isDone)
            {
                Debug.Log("Loading the Scene");
                yield return null;
            }
```

ErikLouwerse commented on Hivemind/Assets/Tests/PlayModeTests/GatherResourceTest.cs:
```diff
@@ -0,0 +1,88 @@
+{
+    public class GatherResourceTest
+    {
+
```
WHITESPACE

ErikLouwerse commented on Hivemind/Assets/Tests/PlayModeTests/GatherResourceTest.cs:
```diff
@@ -0,0 +1,88 @@
+        [UnitySetUp]
+        public IEnumerator Init()
+        {
+            SceneManager.LoadScene("TestScene");
+            yield return new WaitForSeconds(1f);
```
```suggestion
            AsyncOperation asyncLoadLevel = SceneManager.LoadSceneAsync("TestScene", LoadSceneMode.Single);
            while (!asyncLoadLevel.isDone)
            {
                Debug.Log("Loading the Scene");
                yield return null;
            }
```

ErikLouwerse commented on Hivemind/Assets/Tests/PlayModeTests/MindBuilderTest.cs:
```diff
@@ -0,0 +1,118 @@
+﻿using NUnit.Framework;
```
too much bad white space to comment

MMCQ33 committed:
Update Hivemind/Assets/Tests/PlayModeTests/GatherResourceTest.cs

Co-authored-by: ErikLouwerse <32570430+ErikLouwerse@users.noreply.github.com>

MMCQ33 committed:
Update Hivemind/Assets/Tests/PlayModeTests/BaseBuildingTest.cs

Co-authored-by: ErikLouwerse <32570430+ErikLouwerse@users.noreply.github.com>

MMCQ33 committed:
updated Branche to match comments

fixed comments that were left on this branche.

MMCQ33 committed:
Merge branch '#40-Testing' of https://github.com/zoompower/Hivemind into #40-Testing

MMCQ33 committed:
fixed debug logs

fixed debug logs and added last wait until sceneload.

MMCQ33 committed:
Added enter

added enter to make code more readable

MMCQ33 committed:
fixed whitespace in gatheringtest

MMCQ33 committed:
Merge branch 'Develop' into #40-Testing

MMCQ33 committed:
Fixed merge

fixed problems with merge. Fixed ants moving schokingly.

MMCQ33 committed:
Made ant enable navmesh

MMCQ33 committed:
removed unused using

MMCQ33 committed:
removed get set

MMCQ33 committed:
changed location of amount of resrouces

MMCQ33 committed:
Merge branch 'Develop' into #40-Testing

MMCQ33 committed:
Smoothed out some bugs

made ants walk smoother to the outside. Made game resume when you resumed from the pause menu.
